src/hooks/useNavbarToggle.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react"
+
+const useNavbarToggle = () => {
+  const [isToggle, setIsToggle] = useState(false)
+  const toggleRef = useRef<HTMLButtonElement>(null)
+
+  const handleOutSideClick = (e: Event) => {
+    const current = toggleRef.current
+    if (isToggle && current && !current.contains(e.target as Node)) {
+      setIsToggle(false)
+    }
+  }
```

Comment:
[P1] 이 부분 `as` 사용 하신 부분과 조건을 나눠서 정의해주시면 가독성에도 좋고 코드도 안정적일 것 같습니다.

```suggestion
    const handleOutSideClick = useCallback(
    ({ target }: MouseEvent) => {
      const { current } = toggleRef

      if (!isToggle) {
        return
      }

      if (!current || !(target instanceof HTMLElement)) {
        return
      }

      if (!current.contains(target)) {
        setIsToggle(false)
      }
    },
    [isToggle],
  )
```